src/flb_pack.c

```diff
@@ -1512,9 +1512,31 @@ flb_sds_t flb_msgpack_to_gelf(flb_sds_t *s, msgpack_object *o,
                 } else if (v->type == MSGPACK_OBJECT_STR){
                     val     = v->via.str.ptr;
                     val_len = v->via.str.size;
-                    if ( val_len != 1 || val[0] < '0' || val[0] > '7' ) {
+                    if ( !(val_len == 1 && val[0] >= '0' && val[0] <= '7' )) {
+                        char* allowed_levels[] = {
+                            "emerg",
+                            "alert",
+                            "crit",
+                            "err",
+                            "warning",
+                            "notice",
+                            "info",
+                            "debug"
+                        };
+                        const size_t array_size = sizeof(allowed_levels) / sizeof(allowed_levels[0]);
+                        int i;
+                        
+                        for (i = 0; i < array_size; ++i) {
+                            if (!strncmp(mk_string_tolower(val), allowed_levels[i], val_len)) {
```

Comment:
The `mk_string_tolower` does a malloc using `mk_string_dup`, so we are leaking memory, and if I'm not wrong msgpack strings are not null terminated. It's best to use `strncasecmp`

Reply:
Agreed. Changed to `strncasecmp`. I also changed the size of comparison to `allowed_levels[i]` to make it more accurate.
Anyway, this also matches `errooooo`. IMO, it is okay, because I think it's better to be optimistic about logs and ship them as long as we can. But if you think it needs to be exact, I can also check equality of length of strings.